home-work-app/src/app/components/FilterPanel/FilterPanel.tsx

```diff
@@ -1,28 +1,65 @@
 import { SortOptions } from '../SortOptions/SortOptions';
 import { StyledFilterPanel, FilterOptions, FilterOptionItem , SelectedFilterOptionItem} from './FilterPanelStyleSet.styled';
-import { FilterItems, FilterConfigItem } from '../../types';
+import { FilterItems, FilterConfigItem, MapDispatchToProps } from '../../types';
 import React, { useCallback, useState } from 'react';
+import { setUpdateStatus } from '../../redux/actions/setUpdateStatus';
+import { updateMovieAction } from '../../redux/actions/updateMovies';
+import { connect } from 'react-redux';
+import { IMovieActions } from '../../interfaces/IMovieActions';
+
+type RequestParams = {
+    sortBy: string,
+    filter?: string[]
+};
+
+interface FilterPanelProps extends MapDispatchToProps {
+    filterItems: FilterItems;
+}
 
 const sortOptions: string[] = [
     'RELEASE',
     'TITLE',
-    'GENRE',
-    'YEAR'
+    'RATING'
 ];
 
-export const FilterPanel: React.FC<{filterItems: FilterItems}> = (props: {filterItems: FilterItems}) => {
+const sortMap: {
+    [key: string]: string
+} = {
+    release: 'release_date',
+    title: 'title',
+    rating: 'vote_average'
+};
+
+const mapDispatchToProps: MapDispatchToProps = {
+    update: updateMovieAction,
+    setUpdated: setUpdateStatus
```

Comment:
Why do you need to set this flag manually before every filtering or sorting?
I suppose it could be a part of async filter and sort actions.